psl-core/src/test/java/org/linqs/psl/evaluation/statistics/RankingEvaluatorTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * This file is part of the PSL software.
+ * Copyright 2011-2015 University of Maryland
+ * Copyright 2013-2021 The Regents of the University of California
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.linqs.psl.evaluation.statistics;
+
+import static org.junit.Assert.assertEquals;
+
+import org.linqs.psl.application.learning.weight.TrainingMap;
+import org.linqs.psl.database.DataStore;
+import org.linqs.psl.database.Database;
+import org.linqs.psl.database.Partition;
+import org.linqs.psl.database.atom.PersistedAtomManager;
+import org.linqs.psl.database.loading.Inserter;
+import org.linqs.psl.database.rdbms.RDBMSDataStore;
+import org.linqs.psl.database.rdbms.driver.H2DatabaseDriver;
+import org.linqs.psl.model.atom.GroundAtom;
+import org.linqs.psl.model.atom.RandomVariableAtom;
+import org.linqs.psl.model.predicate.StandardPredicate;
+import org.linqs.psl.model.term.Constant;
+import org.linqs.psl.model.term.ConstantType;
+import org.linqs.psl.model.term.UniqueIntID;
+import org.linqs.psl.util.MathUtils;
+
+import org.linqs.psl.util.MathUtils;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class RankingEvaluatorTest extends EvaluatorTest<RankingEvaluator> {
```

Comment:
I think you should favor tests in the style of AUREvaluatorTest (since it is newer) rather than CategoricalEvaluatorTest.

It will take a little more work making an `init()` that works, but the idea is that you can then make many cases easily and test against a reference implementation.
For the AUREvaluatorTest, I just put in some test cases than they generated random numbers and tested against scikit-learn.
I even made it so that my script that generated the numbers and ran sklearn also output the Java code so I could just copy and paste.